test/system/src/main/java/io/pravega/test/system/framework/services/PravegaProperties.java

```diff
@@ -0,0 +1,100 @@
+    private static final ImmutableMap<String, String> DEFAULT_PROPERTIES = ImmutableMap.<String, String>builder()
+            // Segment store properties
+            .put("autoScale.muteInSeconds", "120")
+            .put("autoScale.cooldownInSeconds", "120")
+            .put("autoScale.cacheExpiryInSeconds", "120")
+            .put("autoScale.cacheCleanUpInSeconds", "120")
+            .put("curator-default-session-timeout", "10000")
+            .put("bookkeeper.bkAckQuorumSize", "3")
+            // Controller properties.
+            .put("controller.transaction.maxLeaseValue", "60000")
+            .put("controller.retention.frequencyMinutes", "2")
+            .put("log.level", "DEBUG")
+            .build();
+
+    private final Map<String, String> properties = new HashMap<String, String>(DEFAULT_PROPERTIES);
+    private PravegaProperties(){}
+
+    /**
+     * This method adds a key only if it is absent, assuming the default values never need to change
+     *
+     * */
+    private void addProperty(String key, String value) {
+        properties.putIfAbsent(key, value);
+    }
+
+    public ImmutableMap<String, String> getProperties() {
+        return ImmutableMap.copyOf(properties);
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder strBuilder = new StringBuilder();
+        for (String e: this.getProperties().keySet()) {
+            strBuilder.append(e);
+            strBuilder.append(":");
+            strBuilder.append(this.getProperties().get(e));
+            strBuilder.append("|");
+        }
+        return strBuilder.toString();
+    }
+
+    /**
+     * This class overrides the lombok builder.
+     * Add all methods for building PravegaProperties (with/without Auth/TLS) to this class
+     */
+    public static final class PravegaPropertiesBuilder {
+        private static final PravegaProperties DEFAULT_INSTANCE = new PravegaProperties();
+
+        public static PravegaProperties buildWithSecurity() {
+            log.debug("Security Builder invoked.");
+            PravegaProperties props = new PravegaProperties();
+            addControllerProperties(props);
+            addSegmentStoreProperties(props);
+            return props;
+        }
+
+        public static PravegaProperties build() {
+            log.debug("Default Builder invoked.");
+            return DEFAULT_INSTANCE;
+        }
+
+        private static void addControllerProperties(PravegaProperties props) {
+            //auth
+            props.addProperty("controller.auth.enabled", "true");
+            props.addProperty("controller.auth.userPasswordFile", "/opt/pravega/conf/passwd");
+            props.addProperty("controller.auth.tokenSigningKey", "secret");
+        }
+
+
+        private static void addSegmentStoreProperties(PravegaProperties props) {
+            // auth
+            props.addProperty("autoScale.authEnabled", "true");
+            props.addProperty("autoScale.tokenSigningKey", "secret");
```

Comment:
dont we need to pass password file or some other configuration that enable client to talk to controller?

Reply:
For basic(default) auth, file is already present in the controller docker image at the said location.
For system tests, more complex type of security setups won't be supported. 
See: https://github.com/pbelgundi/pravega-resources/blob/master/Executing_Security_System_Tests.md